(pvt-td-entropy)=
# Энтропия. Второе и третье начала термодинамики
Что такое термическое равновесие, и почему энергия переходит от одного тела к другому? Ранее были изложены понятия температуры, количества тепла, работы с точки зрения нулевого и первого начал термодинамики. Сейчас более подробно остановимся на процессе релаксации, достижения термического равновесия. Для этого необходимо более углубленно подойти к пониманию температуры и тепла, а также ввести новый параметр – *энтропию*.


## Микро- и макросостояния. Статистический вес макросостояния
Для начала необходимо ввести такие понятия как ***микро-*** и  ***макросостояния***. Рассмотрим систему (например, газ), состоящую из некоторого количества элементарных составных частей – частиц (молекул). Каждая частица имеет набор состояний, в которых она может находиться.

```{admonition} Определение
:class: tip
Термодинамическое состояние системы, характеризующееся набором макромасштабных параметров (давление, объем, температура и т.д.), соотвествует множеству возможных состояний частиц системы. Некоторое состояние всех частиц представляет собой ***микросостояние***. В то время как термодинамическое состояние является ***макросостоянием***. Набор микросостояний, дающих одно и то же макросостояние системы, называется ***статистическим весом макросостояния*** (*multiplicity of the macrostate*).
```

То есть статистический вес характеризует набор параметров микросостояния, которые могут соответствовать макросостоянию.


Для понимания рассмотрим пример из статистики. Допустим, имеется два игральных кубика. Сумму значений на двух кубиках будем рассматривать как макросостояние, а значение на каждом отдельном кубике – как микросостояние. Таким образом, существует 11 макросостояний рассматриваемой системы: (2, 3, 4, ..., 11, 12). При этом, каждое макросостояние такой системы может быть выражено в виде отдельных микросостояний. Например, макросостояние, при котором сумма значений на двух кубиках равна 4, может быть представлено тремя различными микросостояниями: (1, 3), (3, 1) и (2, 2). Например, макросостояние 7 может быть представлено шестью различными микросостояниями: (1, 6), (6, 1), (2, 5), (5, 2), (3, 4), (4, 3). Таким образом, для каждого макростостояния у нас имеется набор различных микросостояний:

(макросостояние, статистический вес) = (2, 1), (3, 2), (4, 3), (5, 4), (6, 5), (7, 6), (8, 5), (9, 4), (10, 3), (11, 2), (12, 1).

```{admonition} NB
:class: note
Чтобы найти вероятность получения макросостояния системы, необходимо количество возможных вариантов его получения разделить на общее количество микросостояний системы.
```

То есть вероятность получения макросостояния 4 будет равна $\frac{3}{36} = \frac{1}{12} .$ Вероятность получения макросостояния 7 будет равна $\frac{6}{36} = \frac{1}{6}.$


Представим, что имеется две макросистемы, которые изолированы от откружающей среды, но могут обмениваться энергией между собой. При этом, этот обмен не приводит к нарушению структуры систем на микроуровне. Следующее предположение, которое необходимо сделать, является основным постулатом статистической механики:

```{admonition} Постулат
:class: important
Для изолированной системы, находящейся в термодинамическом равновесии, все микросостояния равновероятны.
```

То есть, фактически, с точки зрения статистической механики (и термодинамики в том числе) для нас нет разницы между микросостояниями (1, 3), (3, 1) и (2, 2). Другими словами, [нулевое](TD-2-ThermalEquilibrium.md#pvt-td-thermalequilibrium-zerolaw) и [первое](TD-3-Heat-Work.md#pvt-td-heatwork-firstlaw) начала термодинамики с точки зрения статистики можно описать следующим образом: *в термическом равновесии системы все ее микросостояния равновероятны*.


Сформулируем ***второе начало термодинамики*** с точки зрения статистики:

<a id='pvt-td-entropy-secondlaw-multiplicity'></a>
```{admonition} Постулат
:class: important
Состояние изолированной термодинамической системы, находящейся в равновесии, характеризуется наибольшим статистическим весом.
```

Из данного выражения следует, что процесс релаксации системы можно представить как процесс, в ходе которого происходит увеличение статистического веса макросостояния до достижения максимального значения. Также необходимо подчеркнуть, что в приведенной выше формулировке важную роль играет термин *изолированная*. Это означает, что рассматриваемая система закрыта по отношению к окружающим системам, и что ее энергия постоянна. Данная формулировка [второго начала термодинамики](#pvt-td-entropy-secondlaw-multiplicity) позволяет установить связь между микроуровневым описанием системы и термодинамическим. Такое статистическое описание системы, когда она представляется закрытой с фиксированной энергией, объемом и количеством частиц, называется *микроканоническим ансамблем*, который, по своей сути, является идеализированным представлением реальной системы. Термин *ансамбль* здесь употребляется для описания гипотетически большого количества микросостояний системы.


Рассмотрим другой пример из статистики. Пусть имеется система, состоящая из $N$ монет. И пусть макросостояние такой системы определяется количеством "орлов" и "решек". Следовательно, существует $2^N$ возможных микросостояний. Пусть $N=4$, тогда количество макросостояний будет равно 5 (ноль "орлов", один "орел", два "орла", три "орла", и четыре "орла"). При этом, статистический вес макростостояния с двумя "орлами" будет равен 6 (то есть 6 различными вариантами можно получить два "орла" на четырех монетах): ООРР, ОРОР, ОРРО, РООР, РОРО, РРОО. В общем виде статистический вес макросостония $N$ монет, на которых имеется $N_H$ орлов рассчитывается следующим образом:

$$ \begin{pmatrix} N \\ N_H \end{pmatrix} = \frac{N!}{N_H! \left( N - N_H \right)!}. $$

Например, для приведенного выше примера:

In [1]:
from math import factorial
N = 4
NH = 2
factorial(N) / (factorial(NH) * factorial(N - NH))

6.0

Приведенная формула является количеством сочетаний из $N$ по $N_H$.


Рассмотрим одноатомный газ. Под внутренней энергией такой системы будем понимать суммарную кинетическую энергию ее молекул. Пусть в данной системе имеется $N$ молекул с массой $m$. Каждая молекула в данной системе будет характеризоваться вектором положения $\mathbf{x_i}, \; i = 1 \ldots N$ (вектор, выходящий из начала координат и оканчивающийся в точке нахождения молекулы в данный момент) и вектором импульса $\mathbf{p_i}, \; i = 1 \ldots N$. В рассматриваемом трехмерном пространстве векторы всех молекул раскладываются на $3N$ проекций. Тогда внутренняя энергия такой системы:

$$ U = \sum_{i=1}^{3N} \frac{\mathbf{p_i}^\top \mathbf{p_i}}{2 m_i}. $$

Таким образом, микросостояния определяются заданием $\left( \mathbf{p_i}, \, \mathbf{p_i} \right), \; i = 1 \ldots N$ для всех молекул системы.


<a id='pvt-td-entropy-idealgasmultiplicity'></a>
Допустим, необходимо рассчитать статистический вес макросостояния системы, определяемого внутренней энергией $U$, количеством частиц $N$ и объемом $V$. При этом, статистический вес макросостояния не будет равен бесконенчности (несмотря на непрерывное изменение векторов положения и импульса частиц). Согласно квантовой механике и ее принципу неопределенности, для идеального одноатомного газа статистический вес макросостояния системы определяется следующим выражением ($h$ – константа Планка) для систем с $N >> 1 $:

$$ \frac{1}{N!} \frac{V^N}{h^{3N}} \frac{\pi^{\frac{3N}{2}}}{\left( \frac{3N}{2}\right)!} \left( 2 m U \right) ^ {\frac{3N}{2}}. $$

В данном разделе перед нами не стоит задача вывода данного выражения и понимания основ квантовой механики. Это выражение приведено здесь в качестве иллюстрации определения статистического веса реальной (пусть и идеализированной) системы моноатомного газа. Понятно, что сам расчет внутренней энергии газа через сумму импульсов всех его молекул достаточно проблематичен, но возможен, например, через среднюю скорость всех молекул газа. Таким образом, понятие *статистического веса* позволяет установить связь между микроуровневым описанием системы и термодинамическим.


Также стоит отметить, что в представленной формуле статистического веса предполагается, что все частицы неразличимы между собой, то есть не существует такого эксперимента, который бы позволил отделить одну молекулу газа от другой. То есть все молекулы одноатомного газа, по сути, одинаковы. Иными словами, микросостояние, при котором молекула $a$ имеет векторы положения и импульса $\left( \mathbf{x_a}, \mathbf{p_a} \right)$, а молекула $b$ имеет векторы положения и импульса $\left( \mathbf{x_b}, \mathbf{p_b} \right)$, считается аналогичным микросостоянию, при котором молекула $a$ имеет векторы положения и импульса $\left( \mathbf{x_b}, \mathbf{p_b} \right)$, а молекула $b$ имеет векторы положения и импульса $\left( \mathbf{x_a}, \mathbf{p_a} \right)$. Данное предположение значительно упрощает расчет статистического веса макросостояния.


Стоит отметить еще одно свойство статистического веса макросостония системы как параметра. Рассмотрим на следующем примере. Допустим, имеется контейнер, содержащий газы $A$ и $B$. Данную систему можно мысленно заменить другой, в которой контейнер разделен на объемы $V_A$ и $V_B$ перегородкой, которая может пропускать энергию. Статистический вес такой системы будет равен произведению статистических весов газов $A$ и $B$:

$$ \Omega = \Omega_A \Omega_B. $$

Здесь можно провести аналогию с рассмотренным ранее примером с кубиками. Допустим, имеется одна система, состоящая из двух кубиков, и другая система, также состоящая из двух кубиков. В качестве макросостояния будем рассматривать сумму значений на кубиках. Пусть на первых двух кубиках сумма значений равна 4, а на втором – 3. Статистический вес первой системы равен 3, а второй – 2. То есть число 4 можно получить тремя различными вариантами, а число 3 – двумя. Тогда общее количество различных комбинаций из данных систем будет равно 6.


Следующий пример, который будет рассмотрен ниже, предназначен для понимания направленности процессов и их возможной обратимости. То есть будет являться иллюстрацией ко [второму началу термодинамики](#pvt-td-entropy-secondlaw-multiplicity). Представим, что имеется контейнер объемом $V$, заполненный идеальным газом с давлением $P$, находящийся в термическом равновесии с окружающей комнатой при температуре $T$. При этом, комната, объем которой в несколько раз больше объема контейнера $\left( nV \right)$, является изолированной от окружающих систем. Теперь представим, что мы откроем данный контейнер. Что произойдет? После достижения равновесия газ, находившийся в контейнере, заполнит всю комнату. При этом, стоит обратить внимание, что никакое тепло не было подведено ни контейнеру, ни комнате, то есть $Q = 0$. Также, поскольку не произошло изменение температуры, то этот процесс является изотермическим $\left( \Delta U = 0 \right)$. Согласно первому началу термодинамики, также нулю будет равна и работа $\left( W = 0 \right)$. То есть после достижения равновесного состояния объем газа увеличился с $V$ до $nV$, при этом, согласно [уравнению состояния идеального газа](TD-1-Basics.md#pvt-td-basics-idealgaseos), давление газа уменьшилось от $P$ до $\frac{P}{n}$. Теперь, зададимся вопросом: можем ли мы обратить данный процесс? То есть обратно поместить все $N$ молекул газа в контейнер? Другими словами, может ли газ снова находиться в равновесном состоянии при объеме $V$, температуре $T$ и давлении $P$. [Первое начало термодинамики](TD-3-Heat-Work.md#pvt-td-heatwork-firstlaw) и [уравнение состояния идеального газа](TD-1-Basics.md#pvt-td-basics-idealgaseos) не отрицают этого. Согласно им, газ, действительно, может находиться в этом равновесном макросостоянии. Однако, почему же этого не произойдет? Попробуем оценить, как изменился статистический вес нашей системы, когда она перешла из одного макросостояния в другое. Согласно приведенной выше формуле, статистический вес исходного макросостояния (когда газ находился в контейнере) был пропорционален $V^N$. После открытия контейнера и достижения равновесного состояния статистический вес стал пропорционален $\left( nV \right)^N$. То есть во втором макросостоянии статистический вес системы в $n^N$ раз больше. Пусть комната в два раза больше контейнера $\left( n = 2 \right)$, а идеальный газ представлен тремя молекулами $\left( N = 3 \right)$. Тогда статистический вес второго макросостояния в 8 раз больше статистического веса первого макросостояния. То есть вероятность нахождения системы во втором макросостоянии в 8 раз больше. Однако эта разница не так значительна. Теперь представим, что система состоит из ста молекул $\left( N = 100 \right)$. Тогда статистический вес второго состояния будет больше уже в $2^{100} \approx 10^{30}$ раз. При этом, сто молекул достаточно сложно назвать макросистемой. Можно считать, что если количество молекул в системе соизмеримо с $10^{23}$, то такую систему можно считать макросистемой. Для такой системы вероятность заполнения ею контейнера фактически равна нулю. Из этого следует, что *свободное* расширение газа по своей сути является необратимым процессом. Напротив, квази-стационарное расширение (или сжатие) газа в цилиндре, при котором совершается работа, является обратимым процессом.


Примером выше было проиллюстрировано [второе начало термодинамики](#pvt-td-entropy-secondlaw-multiplicity) *(равновесное состояние термодинамической изолированной системы характеризуется наибольшим статистическим весом)*.


## Энтропия. Второе начало термодинамики
Статистический вес является важным параметром макросистем, который имеет огромные значения. Поскольку данным параметром не слишком удобно оперировать в ходе расчета, введем новый параметр ***энтропию***:


<a id='pvt-td-entropy-definition'></a>
```{admonition} Определение
:class: tip
***Энтропия*** термодинамической системы прямо пропорциональна логарифму ее статистического веса:

$$ S = k \ln \Omega. $$

```

Здесь $k$ – постоянная Больцмана, а $\Omega$ – статистический вес макросостояния системы. Единицы измерения энтропии – $\frac{Дж}{K}$. При этом, энтропия сохраняет основные "закономерности" статистического веса: чем больше количество молекул в системе, тем больше энтропия. Чем больше энергия системы, тем также больше энтропия.

```{admonition} NB
:class: note
Стоит также отметить, что энтропия является экстенсивным параметром (то есть она обладает свойством аддитивности):

$$ S_{AB} = k \ln \Omega_{AB} = k \ln \left( \Omega_A \Omega_B \right) = k \ln \Omega_A + k \ln \Omega_B = S_A + S_B. $$

```

Поскольку энтропия имеет тот же смысл, что и статистический вес, то ***второе начало термодинамики*** можно сформулировать следующим образом:


<a id='pvt-td-entropy-secondlaw-entropy'></a>
```{admonition} Постулат
:class: important
Состояние термодинамической изолированной системы, находящейся в равновесии, характеризуется наибольшей энтропией.
```


Таким образом, процесс релаксации системы можно интерпретировать как процесс достижения равновесного состояния, сопровождающийся увеличением энтропии системы до достижения максимального значения в равновесном состоянии.


<a id='pvt-td-entropy-idealgasentropy'></a>
Получим выражение для расчета энтропии идеального газа с учетом приведенной выше формулы статистического веса. Энтропия идеального газа:

$$ S \approx k \ln \left( \frac{1}{N!} \frac{V^N}{h^{3N}} \frac{\pi^{\frac{3N}{2}}}{\left( \frac{3N}{2} \right)!} \left( 2 m U \right) ^ {\frac{3N}{2}} \right). $$

С учетом свойств логарифма данное выражение можно преобразовать следующим образом:

$$ \frac{S}{k} = \ln \left( V^N \right) + \ln \left( \frac{2 \pi m U}{h^2} \right)^{\frac{3 N}{2}} - \ln N! - \ln \left( \left( \frac{3 N}{2} \right)! \right). $$

С учетом [аппроксимации Стирлинга](https://en.wikipedia.org/wiki/Stirling%27s_approximation) $\left( \ln n! \approx n \ln n - n \right)$:

$$ \frac{S}{k} = N \left( \ln V - \ln N + \ln \left( \frac{2 \pi m U}{h^2} \right)^\frac{3}{2} - \frac{3}{2} \ln \left( \frac{3 N}{2} \right) + \frac{5}{2} \right). $$

Данное выражение преобразуется к следующему виду:

$$ S = k N \left( \ln \left( \frac{V}{N} \left( \frac{4 \pi m}{3 h^2} \frac{U}{N} \right)^\frac{3}{2} \right) + \frac{5}{2} \right). $$

Это выражение позволяет оценить энтропию моноатомного идельного газа и известно как уравнение [Сакура-Тетрода](https://en.wikipedia.org/wiki/Sackur–Tetrode_equation). При этом, экстенсивные параметры системы (объем и внутренняя энергия) в данном выражении записываются в интенсивной форме (на одну молекулу): $\frac{U}{N}$ и $\frac{V}{N}$. Экстенсивная природа энтропии появляется при умножении на количество молекул в системе. Данное выражение можно использовать для оценки изменения энтропии системы при различных процессах.


Теперь рассмотрим влияние некоторых процессов на изменение энтропии системы. Начнем с расширения. Если энергия и количество молекул системы остаются постоянными, то расширение приводит к увеличению энтропии:

$$ \Delta S = S_2 - S_1 = k N \ln \frac{V_2}{V_1} > 0. $$

Если изотермически подвести газу некоторое количество тепла, то, согласно [первому началу термодинамики](TD-3-Heat-Work.md#pvt-td-heatwork-firstlaw), он совершит работу расширения, и его энтропия увеличится.


При смешивании двух газов также будет происходить изменение энтропии системы. Рассмотрим следующий пример. Допустим, имеется контейнер, разделенный перегородкой пополам. В первой части находится газ $A$, во второй – газ $B$. Что произойдет, когда установится равновесие после того, как перегородка будет удалена? Произойдет смешение газов, при этом каждый газ займет весь объем контейнера. Рассмотрим изменение энтропии каждого из них.

$$ \Delta S_A = k N_A \ln 2; \\ \Delta S_B = k N_B \ln 2. $$

При этом, общее изменение энтропии системы:

$$ \Delta S = \Delta S_A + \Delta S_B = k \left( N_A + N_B \right) \ln 2 > 0. $$

Стоит также отметить, что установившееся равновесное состояние является также *диффузивным равновесием*. Увеличение энтропии в данном случае также сопровождается необратимостью процесса.


Теперь остановимся на вопросе (не)обратимости термодинамических процессов. Согласно [второму началу термодинамики](#pvt-td-entropy-secondlaw-entropy), для изолированной системы энтропия (или статистический вес) максимальна в равновесном состоянии. Это означает, что если происходит некий процесс, в ходе которого изменяется состояние системы, равновесие, к которому будет стремится система, будет ограничено необходимым условием увеличения энтропии в ходе данного процесса. Процессы, которые приводят к увеличению энтропии изолированной системы, как например, рассмотренный ранее пример с расширением газа, являются *необратимыми*. Такие процессы не могут протекать в обратном направлении, поскольку это будет противоречить [второму началу термодинамики](#pvt-td-entropy-secondlaw-entropy). Однако, в принципе, это возможно, но только если энтропия системы будет оставаться постоянной. Такие процессы называются *изоэнтропийными*. Также необходимо отметить, что [второе начало термодинамики](#pvt-td-entropy-secondlaw-entropy) показывает направление протекания процесса для *изолированной* системы. Это вовсе не значит, что энтропия системы не может быть уменьшена. Если система может обмениваться энергией с окружающими системами, то в процессе релаксации энтропия такой системы может уменьшиться.


## Изменение энтропии в термодинамических процессах

Представим, что некоторая изолированная система находится в процессе достижения некоторого равновесного состояния (то есть термического, механического, диффузивного равновесия), при этом, если будем рассматривать данную систему, как состоящую из некоторых подсистем (также макроскопических), то в процессе релаксации логично предположить, что будет происходить обмен энергией, объемом, частиц между данными подсистемами. При этом, равновесное состояние, характеризующееся этими параметрами, будет соответствовать максимальной энтропии системы. Следовательно, исходя из этого можно установить взаимосвязь между энтропией и параметрами системы.


Рассмотрим две системы, изолированные от окружающих систем и находящиеся в термическом равновесии с внутренней энергией $U = U_A + U_B$. Поскольку данные системы находятся в равновесии, то их энтропия максимальна. Из этого следует, что частная производная общей энтропии $S = S_A + S_B$ в экстремуме по внутренней энергии как системы $A$, так и системы $B$ будет равна нулю и, следовательно, равны между собой:

$$ \begin{align}
\left( \frac{\partial S}{\partial U_A} \right)_{N_A, V_A} &=  \left( \frac{\partial S}{\partial U_B} \right)_{N_B, V_B}; \\
\left( \frac{\partial S_A}{\partial U_A} \right)_{N_A, V_A} + \left( \frac{\partial S_B}{\partial U_A} \right)_{N_A, V_A} &= \left( \frac{\partial S_A}{\partial U_B} \right)_{N_B, V_B} + \left( \frac{\partial S_B}{\partial U_B} \right)_{N_B, V_B}; \\
\left( \frac{\partial S_A}{\partial U_A} \right)_{N_A, V_A} &= \left( \frac{\partial S_B}{\partial U_B} \right)_{N_B, V_B}.
\end{align} $$

То есть, в равновесном состоянии приращение энтропии каждой системы при изменении их внутренних энергий равны. Поскольку, согласно [нулевому началу термодинамики](TD-2-ThermalEquilibrium.md#pvt-td-thermalequilibrium-zerolaw), две системы, находящиеся в равновесии, характеризуются равной температурой, то именно температура выступает в качестве численного показателя данного приращения энтропии при изменении внутренней энергии. То есть (с учетом ЕИ):

$$ \frac{1}{T} = \left( \frac{\partial S}{\partial U} \right)_{N, V}. $$

Поскольку в данном конкретном случае мы рассматриваем термическое равновесие, то все остальные параметры (количество молекул, объем и т.д.) необходимо зафиксировать.

Рассмотрим, чему равна частная производная энтропии идеального газа по внутренней энергии. Принимая, что уравнение [Сакура-Тетрода](#pvt-td-entropy-idealgasentropy) описывает зависимость энтропии идеального газа от внутренней энергии, получим:

$$ \left( \frac{\partial S}{\partial U} \right)_{N, V} = \frac{1}{T} = \frac{3}{2} \frac{N k}{U}. $$

Следовательно, внутренняя энергия идеального газа:

$$ U = \frac{3}{2} N k T. $$

[Выражение](#pvt-td-entropy-idealgasmultiplicity) для статистического веса макросостояния идеального газа содержало в себе $U^\frac{3N}{2}$. В общем виде это выражение можно записать через $U^\frac{iN}{2}$. Тогда [энтропия](#pvt-td-entropy-definition):

$$ S = k \ln \left( \alpha U^\frac{i N}{2} \right). $$

С учетом этого частная производная энтропии по внутренней энергии:

$$ \left( \frac{\partial S}{\partial U} \right)_{N, V} = \frac{1}{T} = k \frac{i}{2} \frac{N}{U}. $$

Следовательно, внутренняя энергия:

$$ U = \frac{i}{2} N k T. $$

Теперь получим взаимосвязь энтропии и давления. Для этого рассмотрим изолированный контейнер со свободно движущейся перегородкой внутри. Объем контейнера будет равен сумме объемов его частей $V = V_A + V_B.$ Когда данная система придет к термическому и *механическому* равновесию, обусловленному равенством всех действующих сил, то энтропия системы будет максимальна, то есть – находиться в экстремуме. Следовательно, частные производные общей энтропии системы $S = S_A + S_B$ по изменяющимся объемам частей цилиндра также будут равны нулю и, следовательно, равны между собой. С учетом этого:

$$ \begin{align}
\left( \frac{\partial S}{\partial V_A} \right)_{N_A, U_A} &= \left( \frac{\partial S}{\partial V_B} \right)_{N_B, U_B}; \\
\left( \frac{\partial S_A}{\partial V_A} \right)_{N_A, U_A} + \left( \frac{\partial S_B}{\partial V_A} \right)_{N_A, U_A} &= \left( \frac{\partial S_A}{\partial V_B} \right)_{N_B, U_B} + \left( \frac{\partial S_B}{\partial V_B} \right)_{N_B, U_B}; \\
\left( \frac{\partial S_A}{\partial V_A} \right)_{N_A, U_A} &= \left( \frac{\partial S_B}{\partial V_B} \right)_{N_B, U_B}.
\end{align} $$

То есть в равновесном состоянии частные производные энтропий подсистем по объему равны между собой. В качестве численного показателя приращения энтропии по объему выступает давление. Действительно, рассматриваемая система придет в равновесие, когда выравняются давления подсистем. Тогда с учетом ЕИ получим:

$$ \left( T \frac{\partial S}{\partial V} \right)_{N, U} = P. $$

Частная производная [энтропии идеального газа](#pvt-td-entropy-idealgasentropy) по объему:

$$ \left( T \frac{\partial S}{\partial V} \right)_{N, U} = T \frac{N k}{V} = P. $$

Данное выражение согласуется с [уравнением состояния идеального газа](TD-1-Basics.md#pvt-td-basics-idealgaseos).

<a id='pvt-td-entropy-thermodynamicidentity'></a>
Энтропия, как показатель достижения равновесия (как термического, так и механического), может рассмотриваться в качестве функции других параметров (внутренней энергии, объема). То есть $S = S \left( U, V \right).$ Известно, что [дифференциал функции](https://en.wikipedia.org/wiki/Differential_of_a_function) нескольких переменных может быть записан следующим образом:

$$ df = \left( \frac{\partial f}{\partial x_1} \right)_{x_2, \ldots, x_n} dx_1 + \left( \frac{\partial f}{\partial x_2} \right)_{x_1, x_3, \ldots x_n} dx_2 + ... + \left( \frac{\partial f}{\partial x_n} \right)_{x_1, x_2, \ldots, x_{n-1}} dx_n. $$

С учетом этого дифференциал энтропии можно записать следующим образом:

$$ dS = \left( \frac{\partial S}{\partial U} \right)_V dU + \left( \frac{\partial S}{\partial V} \right)_U dV = \frac{1}{T} dU + \frac{P}{T} dV. $$

Нужно отметить, что данное выражение справедливо для систем с постоянным количеством молекул $N$. Полученное выражение выше можно записать следующим образом:

$$ dU = TdS - PdV. $$

Данное выражение называется *thermodynamic identity*. Оно представляет собой набор уравнений, описывающих взаимосвязь между параметрами термодинамической системы $\left( S, T, U, P, V \right)$. Например, из данного выражения следует, что:

$$ \left( \frac{\partial U}{\partial S}\right)_V = T ; \\ \left( \frac{\partial U}{\partial V}\right)_S = -P. $$

Уравнения *thermodynamic identity* применимы к равновесным состояниям, однако также могут быть применены для описания квази-стационарных процессов. Для таких процессов работа может быть записана как $\delta W = -P dV$. С учетом этого [первое начало термодинамики](TD-3-Heat-Work.md#pvt-td-heatwork-firstlaw):

$$ dU = \delta Q - P dV.$$

Объединяя данное уравнение с *thermodynamic identity*, получим выражение для энтропии в следующем виде:

$$ dS = \frac{\delta Q}{T}. $$

Данное выражение показывает изменение энтропии для квази-стационарного процесса за счет передачи тепла. Из данного выражения следует, что если процесс квази-стационарный и *адиабатический* $\left( \delta Q = 0 \right)$, то он также является *изоэнтропийным*. Несмотря на важность данного определения энтропии, оно имеет ограниченную применимость. Например, для рассмотренных выше примеров (смешивание газов, расширение газа в изолированной комнате), в которых происходило увеличение энтропии без учета теплообмена, данное выражение не может быть применимо для расчета энтропии в процессе релаксации системы. То есть данные примеры не являются квази-стационарными.


Рассмотрим несколько процессов, для которых выполним расчет энтропии. Для начала рассчитаем, как изменилась энтропия системы при полном выкипании одного литра воды (при н.у.) в процессе подвода ей определенного количества тепла. Удельная теплота парообразования воды при давлении $1 \; атм$ составляет $2260 \; \frac{кДж}{кг}$. Принимая, что плотность воды равняется $1000 \; \frac{кг}{м^3}$ при н.у. (то есть масса воды составляет $1 \; кг$), получим количество переданного тепла для полного испарения одного литра воды, равное $2260 \; кДж$. Следовательно, изменение энтропии: 

$$ \Delta S = \frac{Q}{T} \approx 6.06 \; \frac{кДж}{K}.$$

Использованное выше значение удельной теплоты парообразования является табличным значением, полученным из разницы энтальпии системы в двух состояниях. Поскольку в рассматриваемом процессе не проиходит передачи энергии в виде работы, то [энтальпия системы](TD-5-Enthalpy.md#pvt-td-enthalpy-definition), характеризующая количество тепла, переданное системе в изобарном процессе, может быть использована для вычисления количества тепла и энтропии. Для расчета свойств воды существует специальная библиотека *[iapws](https://iapws.readthedocs.io/en/latest/)*.

In [2]:
from iapws import IAPWS97
w = IAPWS97(P=0.101325, x=0)
s = IAPWS97(P=0.101325, x=1)
s.h - w.h

np.float64(2256.5407482377573)

Изменение удельной энтропии $\left( \frac{кДж}{кг K} \right)$ системы:

In [3]:
s.s - w.s

np.float64(6.0476611352628105)

Здесь необходимо отметить следующий важный момент. В данном примере мы рассматриваем выкипание воды в результате передачи ей тепла. Однако существуют другие способы нагрева, например, микроволновое воздействие, при котором происходит кипение (и нагревание) в результате работы. При этом, изменение энтропии системы в данном примере не равно нулю. Поскольку кипение воды в результате микроволнового воздействия не является квази-стационарным процессом, то данное выражение не может быть применимо для расчета изменения энтропии системы.


Рассмотрим следующий пример, в ходе которого определим изменение энтропии при смешивании горячей воды объемом $50 \; л$ при температуре $55 \; ^{\circ} C$ и холодной воды объемом $25 \; л$ холодной воды при температуре $10 \; ^{\circ} C$.


Для начала определим массы (в кг) первой (горячей воды) и второй (холодной воды) подсистем:

In [4]:
T1 = 55 + 273.15
m1 = 50 * 10**(-3) / IAPWS97(P=0.101325, T=T1).v
m1

np.float64(49.28535037123039)

In [5]:
T2 = 10 + 273.15
m2 = 25 * 10**(-3) / IAPWS97(P=0.101325, T=T2).v
m2

np.float64(24.99253850423755)

Будем считать данную систему, состоящую из двух подсистем, изолированной от окружающих систем. Поскольку две подсистемы находятся при различных температурах, то будет происходить теплообмен. По определению [теплоемкости](TD-4-HeatCapacity.md#pvt-td-heatcapacity-definition), количество тепла, переданное системе, определяется следующим выражением (поскольку процесс происходит при постоянном давлении, то может использоваться величина изобарной теплоемкости):

$$ \delta Q = C_P dT. $$

Поскольку рассматриваемая система является изолированной, то количество переданного тепла по модулю одной системой будет равняться количеству полученного тепла другой системой. Следовательно, можно записать следующее выражение:

$$ - \int_{T_1}^{T_f} C_P dT = \int_{T_2}^{T_f} C_P dT.$$

Здесь, перед интегралом в левой части уравнения, стоит знак "минус", поскольку первая подсистема отдает количество теплоты. Данное выражение через удельные (на единицу массы) теплоемкости:

$$m_1 \int_{T_1}^{T_f} c_P dT + m_2 \int_{T_2}^{T_f} c_P dT = 0.$$

Для решения данного уравнения (нахождения конечной температуры $T_f$) создадим функцию, которая рассчитывает величину изобарной теплоемкости воды при изменении температуры:

In [6]:
cp = lambda T: IAPWS97(P=0.101325, T=T).cp

Например, при $0 \; ^{\circ} C$ в $\frac{кДж}{кг}$:

In [7]:
cp(273.15)

np.float64(4.219430340168217)

Для нахождения интеграла и решения нелинейного уравнения будем использовать численные методы, реализованные в библиотеке *[scipy](https://docs.scipy.org/doc/scipy/reference/index.md)*. Для нахождения решения нелинейного уравнения необходимо создать функцию, представляющую собой решаемое уравнение.

In [8]:
from scipy import integrate
from scipy.optimize import fsolve
eq = lambda Tf: m1 * integrate.quad(cp, T1, Tf)[0] + m2 * integrate.quad(cp, T2, Tf)[0]
Tf = fsolve(eq, 40+273.15)[0]
Tf - 273.15

np.float64(39.84847730352402)

Получим выражение для расчета изменения энтропии системы:

$$ dS = \frac{\delta Q}{T} = \frac{C_p dT}{T}. $$

Следовательно, для первой и второй подсистем:

$$ \Delta S_1 = \int_{T_1}^{T_f} \frac{C_p}{T}dT = m_1 \int_{T_1}^{T_f} \frac{c_p}{T}dT; \\ \Delta S_2 = \int_{T_2}^{T_f} \frac{C_p}{T}dT = m_2 \int_{T_2}^{T_f} \frac{c_p}{T}dT. $$

С учетом этого энтропия системы:

$$ \Delta S = \Delta S_1 + \Delta S_2.$$

In [9]:
cp_over_T = lambda T: cp(T) / T
dS1 = m1 * integrate.quad(cp_over_T, T1, Tf)[0]
dS2 = m2 * integrate.quad(cp_over_T, T2, Tf)[0]
dS1, dS2, dS1 + dS2

(np.float64(-9.737069606842374),
 np.float64(10.479234773509713),
 np.float64(0.7421651666673394))

При этом, изменение энтропии первой системы (горячей воды) отрицательно, что не противоречит [второму началу термодинамики](#pvt-td-entropy-secondlaw-entropy), поскольку оно применимо для изолированной системы. Изменение энтропии всей системы положительно: $0.742 \; \frac{кДж}{K}.$ Полученные значения соответствуют изменениям энтропии, рассчитанным с использованием библиотеки *[iapws](https://iapws.readthedocs.io/en/latest/)*.

In [10]:
S1 = IAPWS97(P=0.101325, T=T1).s
S2 = IAPWS97(P=0.101325, T=T2).s
S = IAPWS97(P=0.101325, T=Tf).s
(S - S1) * m1, (S - S2) * m2, (S - S1) * m1 + (S - S2) * m2

(np.float64(-9.737069606842136),
 np.float64(10.479234773509885),
 np.float64(0.7421651666677498))

(pvt-td-entropy-thirdlaw)=
## Третье начало термодинамики
В предыдущей задаче было использовано следующее выражение для расчета изменения энтропии системы:

$$ \Delta S = \int_{T_i}^{T_f} \frac{C}{T} dT. $$

Данное выражение справедливо для квази-стационарных процессов. Если допустить, что теплоемкость системы примерно постоянна, то ее можно вынести за знак интеграла. Тогда:

$$ \Delta S = C \int_{T_i}^{T_f} \frac{dT}{T} = C \ln \frac{T_f}{T_i}. $$

Эта формула может использоваться для квази-стационарных процессов при расчете изменения энтропии, обусловленных нагревом или охлаждением системы.


Если допустить, что конечная температура $T_f$ стремится к нулю, то тогда изменение энтропии не может быть определено. Но при этом, изменение энтропии должно быть конечным значением, исходя из определения через статистический вес системы. Исходя из данного противоречия можно сформулировать ***третье начало термодинамики***: *при $T \rightarrow 0$ имеем $S \rightarrow 0$ и $C \rightarrow 0$.* Иными словами, третье начало термодинамики можно сформулировать как:

```{admonition} Постулат
:class: important
Абсолютный ноль температуры недостижим.
```

Выше были рассмотрены процессы, которые являются квази-стационарными. Но как рассчитывать энтропию системы для не квази-стационарных процессов? При комбинировании [thermodynamic identity](#pvt-td-entropy-thermodynamicidentity) с [первым началом термодинамики](TD-3-Heat-Work.md#pvt-td-heatwork-firstlaw) получим:

$$ dU = TdS - PdV; \\ dU = \delta Q + \delta W; \\ TdS = \delta Q + \delta W + PdV. $$

Если рассматриваемый процесс квази-стационарный, то $\delta W = - PdV$, и тогда получим ранее выведенное выражение для дифференциала энтропии. Для не квази-стационарных процессов $\delta W \geq - PdV $, следовательно:

$$ dS \geq \frac{\delta Q}{T}. $$